### ISSUE TITLE
Caching Improvements

### ISSUE THREAD

**Issue by yjperez** [gravitational/teleport#5046]
### Feature Request
Requesting for the ability to reset the caching of nodes in the proxy or a parameter to set ttl for cached items 

### Motivation
Currently, if we destroy an EC2 instance and create another instance with the same hostname when one attempts to ssh to that node via hostname we get an error "Ambiguous host could match multiple nodes hostnames". This intermittently happens for up to 24 hours in some cases. If we have the ability to clear the cache this will be mitigated.

### Who's it for?
OSS User, Pro, Enterprise

**Comment by russjones** [comment]
@benarent The workaround for this is to manually remove the node `tctl rm node/...` but is there a better UX around this?

**Comment by jdconti** [comment]
@russjones I've tried the workaround here and it doesn't seem to work; am I missing something?

```
# tctl nodes ls |grep -i foo
foo                    439c6592-e26e-4b73-b090-714dcf7bb7d9 10.100.100.100:3022   foo=bar
# tctl rm node/foo
ERROR: "/nodes/default/foo" is not found
```

I've tried with the node name and uuid.

**Comment by yjperez** [comment]
@jdconti what version of Teleport are you running? Try running the command in debug by adding `-d` (ex. `tctl rm node/<uuid> -d`) 

This how I removed it on Teleport 6.0.0.
```
[root@proxy-auth ~]# tctl nodes ls | grep k8
k8t-holmes 11820ba1-58a5-4e05-83e7-17461c019be0 x.x.x.x:3022  hostname=k8t-holmes,role=k8-worker
k8-upton   aa1bccf8-5e9c-4d95-909f-ddb0540952cb x.x.x.x:3022  hostname=k8-upton,role=k8-worker
k8-moss    fa0e95a3-f50b-4f8a-92d5-21f6d3302735 x.x.x.x:3022  hostname=k8-moss,role=k8-master
[root@proxy-auth ~]# tctl rm node/11820ba1-58a5-4e05-83e7-17461c019be0
node 11820ba1-58a5-4e05-83e7-17461c019be0 has been deleted
[root@proxy-auth ~]# tctl nodes ls | grep k8
k8-upton   aa1bccf8-5e9c-4d95-909f-ddb0540952cb x.x.x.x:3022  hostname=k8-upton,role=k8-worker
k8-moss    fa0e95a3-f50b-4f8a-92d5-21f6d3302735 x.x.x.x:3022  hostname=k8-moss,role=k8-master
```

**Comment by jdconti** [comment]
@yjperez `6.1.2` and I've confirmed that uuid key is no longer in etcd so it appears cache-related.  I caught up with @russjones on this earlier and it may be related to some other issues were seeing...

**Comment by Erick-Reyes** [comment]
Hey @yjperez it looks like this FR can be close as it's been resolve in PR 9722 in release V8.1.1. Are we good to close this issue?

**Comment by yjperez** [comment]
Closing as resolved in [#9722](https://github.com/gravitational/teleport/pull/9722)
